modules/plottinggl/processors/colorscalelegend.cpp

```diff
@@ -0,0 +1,282 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2018 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/plottinggl/processors/colorscalelegend.h>
+#include <modules/opengl/texture/textureutils.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo ColorScaleLegend::processorInfo_{
+    "org.inviwo.ColorScaleLegend",  // Class identifier
+    "Color Scale Legend",           // Display name
+    "Plotting",                     // Category
+    CodeState::Stable,              // Code state
+    Tags::GL,                       // Tags
+};
+const ProcessorInfo ColorScaleLegend::getProcessorInfo() const { return processorInfo_; }
+
+ColorScaleLegend::ColorScaleLegend()
+    : Processor()
+    , inport_("inport")
+    , outport_("outport")
+    , volumeInport_("volumeInport")
+    , isotfComposite_("isotfComposite", "TF & Isovalues")
+    , positioning_("positioning", "Positioning & Size")
+    , style_("style", "Style")
+    , legendPlacement_("legendPlacement", "Legend Placement",
+                       {{"top", "Top", 0},
+                        {"right", "Right", 1},
+                        {"bottom", "Bottom", 2},
+                        {"left", "Left", 3},
+                        {"custom", "Custom", 4}},
+                       1)
+    , rotation_("legendRotation", "Legend Rotation",
```

Comment:
consider using an enum class for rotation instead of values `0, 1, 2, 3` standing for 0, 90, 180, 270 deg rotation

Reply:
Nä